### PR TITLE
backend: patron_exists implemented

### LINKED ISSUE #323: Implement `patron_exists` method
Currently the method returns `True`. We need:

- [x] Check if patron id provided is one of the users in our system

### PR DESCRIPTION
**Pull request by topless**
* implemented patron exists
* removed unused imports
* fixed a bug with undefined params
* closes #323

### CODE REVIEW

File: tests/api/test_apis.py [hunk 1]
```diff
@@ -46,3 +47,9 @@ def test_item_exists(testdata):
 def test_circulation_item_not_exist(testdata):
     """Test return False if item does not exist."""
     assert not item_exists("not-existing-item-pid")
+
+
+def test_patron_exists(testdata):
+    """Test return True if item exists."""
+    first_patron_pid = testdata["loans"][0][Patron.pid_field]
```
Comment (ntarocco) [inline tests/api/test_apis.py]: Patron is not a real record, so I think `Patron.pid_field` attribute does not exist (unless you add it).

I would also add a test for the negative, when it does not exist.

File: invenio_app_ils/api.py [hunk 2]
```diff
@@ -53,4 +54,10 @@ def item_exists(item_pid):
...
 def patron_exists(patron_pid):
     """Return True if the Patron exists given a PID."""
-    return True
+    try:
+        patron = User.query.filter_by(id=patron_pid).first()
+        if patron:
+            return True
+    except PersistentIdentifierError as ex:
+        return False
+    return False
```
Comment (zzacharo) [inline invenio_app_ils/api.py]: Isn't this redundant? If it doesn't find something it will raise....

File: invenio_app_ils/api.py [hunk 3]
```diff
@@ -53,4 +54,10 @@ def item_exists(item_pid):
 
 def patron_exists(patron_pid):
     """Return True if the Patron exists given a PID."""
-    return True
+    try:
+        patron = User.query.filter_by(id=patron_pid).first()
+        if patron:
+            return True
+    except PersistentIdentifierError as ex:
```
Comment (ntarocco) [inline invenio_app_ils/api.py]: This exception will be never thrown. You might have an SQLAlchemy exception, but with `first()` it will return None if not found.

Maybe something similar to:
```
def patron_exists(patron_pid):
    return User.query.filter_by(id=patron_pid).first() != None
```

**Review by ntarocco — CHANGES_REQUESTED** [review]

File: invenio_app_ils/api.py (hunk 3 above)
Comment (zzacharo) [reply invenio_app_ils/api.py]: This resolves my comment 👍

File: tests/api/test_apis.py (hunk 1 above)
Comment (zzacharo) [reply tests/api/test_apis.py]: Yes you are right :)

Comment (topless) [reply tests/api/test_apis.py]: All done thanks

File: invenio_app_ils/api.py (hunk 3 above)
Comment (topless) [reply invenio_app_ils/api.py]: 👍

**Comment by topless** [comment]
Tests fail also on travis as they do locally.
`tests/api/conftest.py:144:`

`DetachedInstanceError: Instance <Transaction at 0x7fcfeeb34b10> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/bhk3)`

I took the initiative and removed unused imports from `test_mails.py`, it seems if I revert them the tests are passing, @ntarocco @zzacharo @equadon @vladbmihaescu anyone any thoughts or ideas? I don't think its all the dependencies but there is some magic going on there, I think the import of `current_app` is affecting it, will investigate a bit to see whats going on.

So it is the `from invenio_mail.tasks import send_email` that needs to be imported in test_mails so test_apis do not fail.

**Review by zzacharo — APPROVED** [review]

File: invenio_app_ils/api.py [hunk 4]
```diff
@@ -53,4 +54,4 @@ def item_exists(item_pid):
 
 def patron_exists(patron_pid):
     """Return True if the Patron exists given a PID."""
-    return True
+    return User.query.filter_by(id=patron_pid).first() is not None
```
Comment (equadon) [inline invenio_app_ils/api.py:57]: Should we refer to the patron ID as a PID even if it's not a record?

**Review by equadon — APPROVED** [review]

File: invenio_app_ils/api.py (hunk 4 above)
Comment (ntarocco) [reply invenio_app_ils/api.py:57]: Agree with you, the param name comes from `invenio-circulation` that does not know if patron is a record or not. Given that, I would probably not change the "signature" of the method. WDYT?

**Review by ntarocco — APPROVED** [review]
